lib/inc/internal/facts/freebsd/dmi_resolver.hpp

```diff
@@ -0,0 +1,28 @@
+/**
+ * @file
+ * Declares the FreeBSD Desktop Management Interface (DMI) fact resolver.
+ */
+#pragma once
+
+#include "../resolvers/dmi_resolver.hpp"
+
+namespace facter { namespace facts { namespace freebsd {
+
+    /**
+     * Responsible for resolving DMI facts.
+     */
+    struct dmi_resolver : resolvers::dmi_resolver
+    {
+     protected:
+        /**
+         * Collects the resolver data.
+         * @param facts The fact collection that is resolving facts.
+         * @return Returns the resolver data.
+         */
+        virtual data collect_data(collection& facts) override;
+
+     private:
+        std::string kenv_lookup(const char* file);
```

Comment:
I meant something slightly different (kenv_lookup doesn't need to be part of the dmi_resolver class at all, it could be a local function in dmi_resolver.cc) but it doesn't matter.